articles/how-to-consume-data-from-an-API-using-react/index.md

```diff
@@ -0,0 +1,424 @@
+   "Plot":"The Guardians struggle to keep together as a team while dealing with their personal family issues, notably Star-Lord's encounter with his father the ambitious celestial being Ego.",
+   "Language":"English",
+   "Country":"USA",
+   "Awards":"Nominated for 1 Oscar. Another 15 wins & 57 nominations.",
+   "Poster":"https://m.media-amazon.com/images/M/MV5BNjM0NTc0NzItM2FlYS00YzEwLWE0YmUtNTA2ZWIzODc2OTgxXkEyXkFqcGdeQXVyNTgwNzIyNzg@._V1_SX300.jpg",
+   "Ratings":[
+      {
+         "Source":"Internet Movie Database",
+         "Value":"7.6/10"
+      },
+      {
+         "Source":"Rotten Tomatoes",
+         "Value":"85%"
+      },
+      {
+         "Source":"Metacritic",
+         "Value":"67/100"
+      }
+   ],
+   "Metascore":"67",
+   "imdbRating":"7.6",
+   "imdbVotes":"562,796",
+   "imdbID":"tt3896198",
+   "Type":"movie",
+   "DVD":"N/A",
+   "BoxOffice":"$389,813,101",
+   "Production":"Marvel Studios, Walt Disney Pictures",
+   "Website":"N/A",
+   "Response":"True"
+}
+```
+
+In the above `JSON` response, we have data like the `Title,` `Year,` `Rated,` `Released,` `Runtime,` `Genre`, etc, of the movie Guardians of the Galaxy Vol. 2.
+
+### Step 2: Creating a react project.
+
+Let's create a React application using the following command:
+
+```
+npx create-react-app OMDbApi
+```
+
+Once you run the above statement, it'll take a few minutes to install the required dependencies. Once the project is created, you can open the folder in your preferred code editor.
+
+### Step 3: Project Structure
+
+All the files that we'll be working with are located in the `src` folder. These files will be compiled and shown to the user.
+
+The image below shows the overall structure of the React application.
+
+![structure](/engineering-education/how-to-consume-data-from-an-API-using-react/structure.png)
+
+### Step 4: Developing the react application
+
+Use `npm start` to start the development server. By default, the react projects are usually hosted at `http://localhost:3000/`.
+
+We'll write all of our code in the `App.js`. Modify the `App.js` file to look like this before we start.
+
+```JSX
+import React from 'react';
+
+function App() {
+  return (
+    <div className="App">
+    
+    </div>
+  );
+}
+
+export default App;
+```
+
+We should import `useEffect` and `useState` into this component.
+
+```JSX
+import React, {useEffect, useState} from 'react';
+```
+
+The `useEffect` hook allows `React` to perform a specific action when the state changes. In our case, we will use `useEffect` to make network requests to the API when the component mounts for the first time.
+
+The `useState` hook will help in creating a state and a function to update that state. When you update a state, the component will re-render.
+
+Let's create a state for the movie. The `movie` state will hold data returned from the OMDb API. The `setMovie` helps in updating the `movie` state.
+
+```JSX
+const [movie, setMovie] = useState([]);
```

Comment:
why is the initial state an array? it should be an empty object because the response is an object.